opennebula/helpers_vr.go

```diff
@@ -101,6 +101,13 @@ func vrNICAttach(ctx context.Context, timeout time.Duration, controller *goca.Co
 		updatedNICsLoop:
 			for i, nic := range updatedNICs {
 
+				// For VRouter NICs, floating IPs are set using the "IP" field, but it is represented in the ON API
```

Comment:
unless I miss something it seems that this whole code part in helpers_vr.go is useless ?

Reply:
These few lines enable the IP to be set on the NIC resource for vrouter. On VRouter NICs that use floating IP, ON represents the IP in the `VROUTER_IP` field.
This loop is ran when updating NICs. Without this, when running `terraform apply` to update the NIC would yield an empty IP (and therefore trigger a recreation when not necessary)